xbstrap/base.py

```diff
@@ -2614,6 +2615,33 @@ def run_tool_task(cfg, task):
     )
 
 
+def discard_src(cfg, src):
+    with suppress(Exception):
+        shutil.rmtree(os.path.join(cfg.source_root, src.source_subdir))
+
+
+def discard_pkg(cfg, pkg):
+    with suppress(Exception):
+        shutil.rmtree(os.path.join(cfg.build_root, pkg.build_subdir))
+        shutil.rmtree(os.path.join(cfg.build_root, pkg.collect_subdir))
+
+    if cfg.use_xbps:
+        output = subprocess.DEVNULL
+        if verbosity:
+            output = None
+        effective_arch = pkg.architecture
+        environ = os.environ.copy()
+        _util.build_environ_paths(
+            environ, "PATH", prepend=[os.path.join(_util.find_home(), "bin")]
+        )
+        environ["XBPS_ARCH"] = effective_arch
+        args = ["xbps-remove", "-Fy", "-r", cfg.sysroot_dir, pkg.name]
+        _util.log_info("Running {}".format(args))
+        subprocess.call(args, env=environ, stdout=output)
+    else:
+        raise GenericError("Package management configuration does not support discard")
```

Comment:
but it does - you already delete stuff when this happens?

Reply:
+1